dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/InterpretationController.java

```diff
@@ -342,10 +407,16 @@ public void updateComment( @PathVariable( "uid" ) String uid, @PathVariable( "cu
                 }
 
                 comment.setText( content );
+                Set<User> users = interpretationService.getMentionedUsers( content ); 
```

Comment:
Before you commented that about those 3 instances of getMentionedUsers + setMentions + updateSharingForMentions + save/update + sendNotifications. I guess that to be abstracted we should have an interface for interpretation/comment (mentionable) so we can pass it as an argument. Also, it will require a check of the obj.getId() to see if it should be updated or saved. Anyway, I also don't see it as a priority.

Reply:
Yes, I agree this could be an option. They have a similar approach for "interpretation objects" (reports, charts, etc). An abstract class with some common functionality. Let's keep it in mind and maybe we can add it in the next iteration.